src/bases/D3MPoolBase.sol

```diff
@@ -0,0 +1,114 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+// Copyright (C) 2021-2022 Dai Foundation
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+pragma solidity 0.6.12;
+
+interface TokenLike {
+    function approve(address, uint256) external returns (bool);
+    function transfer(address, uint256) external returns (bool);
+    function transferFrom(address, address, uint256) external returns (bool);
+    function balanceOf(address) external view returns (uint256);
+}
+
+interface CanLike {
+    function hope(address) external;
+    function nope(address) external;
+}
+
+interface d3mHubLike {
+    function vat() external view returns (address);
+}
+
+abstract contract D3MPoolBase {
+    // --- Auth ---
+    mapping (address => uint256) public wards;
+    function rely(address usr) external auth {
+        wards[usr] = 1;
+        emit Rely(usr);
+    }
+    function deny(address usr) external auth {
+        wards[usr] = 0;
+        emit Deny(usr);
+    }
+    modifier auth {
+        require(wards[msg.sender] == 1, "D3MPoolBase/not-authorized");
+        _;
+    }
+
+    TokenLike   public immutable asset; // Dai
+    address     public immutable hub;
+    address     public immutable pool;
+
+    address     public           share;
+    uint256     public           live = 1;
+
+    // --- Events ---
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event Cage();
+    // --- EIP-4626 Events ---
+    event Deposit(address indexed caller, address indexed owner, uint256 assets, uint256 shares);
```

Comment:
We currently do not emit these Deposit and Withdraw events anywhere. I think we can just remove them as the hub emits events with the amounts on `wind` and `unwind`.

Reply:
I added these because they were in the 4626 spec.  We could add them to the Pool implementations or because we have the hub events and can assume any 4626 integration we are working with would emit these, we can do as you suggest and remove them.  Do you see a benefit to keeping them and being a little closer to 4626?

Reply:
Personally I don't see much added value as we are not completely compatible with 4626 anyway. I think we can remove as we have the hub events.